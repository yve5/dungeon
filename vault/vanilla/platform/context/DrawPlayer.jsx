export const drawPlayer = (context, platform) => {
  if (platform?.map?.player) {
    const {
      map: { player, tileSize },
      camera,
    } = platform;

    context.fillStyle = player.color;
    context.beginPath();

    context.arc(
      player.loc.x + tileSize / 2 - camera.x,
      player.loc.y + tileSize / 2 - camera.y,
      tileSize / 2 - 1,
      0,
      Math.PI * 2
    );

    context.fill();
  }
};
