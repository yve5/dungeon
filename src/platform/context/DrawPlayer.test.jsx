import { drawPlayer } from './DrawPlayer';

describe('drawPlayer', () => {
  it('should draw player. undefined', () => {
    const fakeContext = new canvas2dContext();

    drawPlayer(fakeContext, undefined);

    expect(fakeContext).toMatchSnapshot();
  });

  it('should draw player. default', () => {
    const fakeContext = new canvas2dContext();

    drawPlayer(fakeContext, {
      camera: {
        x: 16,
        y: 16,
      },
      map: {
        data: [
          [2, 2, 2, 2, 2],
          [2, 1, 1, 1, 2],
          [2, 1, 1, 1, 2],
          [2, 1, 1, 1, 2],
          [2, 2, 2, 2, 2],
        ],
        tileSize: 16,
        player: {
          color: '#FF9900',
          loc: {
            x: 16,
            y: 16,
          },
        },
      },
    });

    expect(fakeContext).toMatchSnapshot();
  });
});
