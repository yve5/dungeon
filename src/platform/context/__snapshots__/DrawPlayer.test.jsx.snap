// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawPlayer should draw player. default 1`] = `
canvas2dContextMock {
  "fillStyle": "#FF9900",
  "logs": Array [
    Object {
      "type": "beginPath",
    },
    Object {
      "args": Array [
        8,
        8,
        7,
        0,
        6.283185307179586,
      ],
      "fillStyle": "#FF9900",
      "type": "arc",
    },
    Object {
      "fillStyle": "#FF9900",
      "type": "fill",
    },
  ],
}
`;

exports[`drawPlayer should draw player. undefined 1`] = `
canvas2dContextMock {
  "fillStyle": "",
  "logs": Array [],
}
`;
